### ISSUE TITLE
Satisfy requirements for inclusion in upstream SWIG

### ISSUE THREAD

**Issue by youngmit** [sethrj/swig#20]
The SWIG docs provide some prerequisites for adding a new language module to the SWIG distribution. See: http://www.swig.org/Doc3.0/SWIGDocumentation.html#Extending_prerequisites

 - [x] Demonstrate basic C code working by porting the "simple" example including a runtime test, see for example Examples/python/simple .
 - [x] Demonstrate basic C++ code working by porting the "class" example including a runtime test, see for example Examples/python/class .
 - [x] Modify configure.ac, Makefile.in and Examples/Makefile.in to run these examples. Please make sure that if the new language is not installed properly on a box, make -k check should still work by skipping the tests and examples for the new language module.
 - [x] Get the test-suite running for the new language (make check-[lang]-test-suite). While the test-suite tests many corner cases, we'd expect the majority of it to work by compiling the generated code correctly as most of the corner cases are covered in the SWIG core. Get at least one C and one C++ runtime test running in the test-suite.
 - [x] Provide a chapter in the html documentation on the basics of using the language module.
Ensure your source code is formatted according to the coding style guidelines.
 - [x] Finally, email the SWIG developers with a patch and a demonstration of commitment to maintaining the language module, certainly in the short term and ideally long term.

Once accepted into the official Git repository, development efforts should concentrate on getting the entire test-suite to work with plenty of runtime tests. Runtime tests should be for existing testcases and new test cases should be added should there be an area not already covered by the existing tests.

**Comment by sethrj** [comment]
OK done, submitted in https://github.com/swig/swig/pull/1195 and waiting on upstream attention
